src/test/user.spec.js

```diff
@@ -456,3 +457,123 @@ describe('PATCH /api/v1/unfollow', () => {
+    });
+
+    it('should update user profile', async () => {
+      const firstUser = await createUser();
+      const numberOfUsers = await BaseRepository.findAll(db.User, {
+        id: firstUser.id
+      });
+      const token = helper.jwtSigner(firstUser);
+      expect(numberOfUsers.length).to.equal(1);
+
+      const res = await server()
+        .put(`${USER_API}/${firstUser.id}`)
+        .set('token', token)
+        .send({ bio: 'I am Nkechi', avatar: 'anythingdotimage.jpg' });
+
+      expect(res.body.message).to.equal('Record successfully updated');
+      expect(res.status).to.equal(200);
+
+      const userProfile = await BaseRepository.findItAll(db.User, {
+        id: firstUser.id
+      });
+
+      expect(userProfile[0].bio).to.equal('I am Nkechi');
+      expect(userProfile[0].avatar).to.equal('anythingdotimage.jpg');
+    });
+  });
+});
+
+describe('PATCH /api/v1/auth/user/:id/role', () => {
+  beforeEach(async () => {
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+
+  it(`should change a user's role to admin`, async () => {
+    const theUser = await createUser();
+    const getSuperAdmin = await getUser();
+    getSuperAdmin.role = 'superadmin';
+    const superAdmin = await createUser(getSuperAdmin);
+
+    const numberOfUsers = await BaseRepository.findItAll(db.User);
+    expect(numberOfUsers.length).to.equal(2);
+    expect(numberOfUsers[0].role).to.equal('user');
+    expect(numberOfUsers[1].role).to.equal('superadmin');
+
+    const token = helper.jwtSigner(superAdmin);
+
+    const res = await server()
+      .patch(`${USER_API}/${theUser.id}/role`)
+      .set('token', token)
+      .send({ role: 'admin' });
+    expect(res.status).to.equal(200);
+
+    const userRoleStatus = await BaseRepository.findItAll(db.User, {
+      id: theUser.id
+    });
+    const currentNumberOfUsers = await BaseRepository.findItAll(db.User);
+
+    expect(userRoleStatus[0].role).to.equal('admin');
+    expect(currentNumberOfUsers.length).to.equal(2);
+  });
+
+  it('should not change role if it is not a superadmin', async () => {
+    const fistUser = await createUser();
+    const secondUser = await createUser();
+
+    const numberOfUsers = await BaseRepository.findItAll(db.User);
+    expect(numberOfUsers.length).to.equal(2);
+    expect(numberOfUsers[0].role).to.equal('user');
+    expect(numberOfUsers[1].role).to.equal('user');
+
+    const token = helper.jwtSigner(fistUser);
+
+    const res = await server()
+      .patch(`${USER_API}/${secondUser.id}/role`)
+      .set('token', token)
+      .send({ role: 'admin' });
+    expect(res.status).to.equal(403);
+
+    const userRoleStatus = await BaseRepository.findItAll(db.User, {
+      id: secondUser.id
+    });
+    const currentNumberOfUsers = await BaseRepository.findItAll(db.User);
+
+    expect(userRoleStatus[0].role).to.not.equal('admin');
+    expect(userRoleStatus[0].role).to.equal('user');
```

Comment:
😍 😍 😍 😍 😍 

I love the logic of this test.

Please remove the superfluous assertions as discussed.